pymatgen/io/shengbte.py

```diff
@@ -0,0 +1,275 @@
+# pymatgen>pymatgen>io
+
+# coding: utf-8
+# Copyright (c) Pymatgen Development Team.
+# Distributed under the terms of the MIT License
+
+
+import f90nml
+import numpy as np
+import re
+
+"""
+This module defines IO ShengBTE for reading, updating, and writing the
+CONTROL input file
+"""
+
+__author__ = "Rees Chang"
+__copyright__ = "Copyright 2019, The Materials Project"
+__version__ = "0.1"
+__email__ = "rc564@cornell.edu"
+__date__ = "June 27, 2019"
+
+
+class ShengBTE_CONTROL_IO:
+
+    """
+    Class for reading, updating, and writing ShengBTE CONTROL files.
+    Currently only supports ShengBTE options relevant to CSLD.
+    """
+
+    def read_CONTROL(self, filename):
+        """
+        Read a CONTROL namelist file and output a namelist object
+        :param filename: Name of the CONTROL file if in current directory.
+            If not, use full path.
+        :return: Dictionary of CONTROL parameters.
+        """
+        nml = f90nml.read(filename)
+        sdict = nml.todict()
+        if 't' in sdict['parameters']:
+            sdict['parameters']['T'] = sdict['parameters']['t']
+            del sdict['parameters']['t']
+        if 't_min' in sdict['parameters']:
+            sdict['parameters']['T_min'] = sdict['parameters']['t_min']
+            del sdict['parameters']['t_min']
+        if 't_max' in sdict['parameters']:
+            sdict['parameters']['T_max'] = sdict['parameters']['t_max']
+            del sdict['parameters']['t_max']
+        if 't_step' in sdict['parameters']:
+            sdict['parameters']['T_step'] = sdict['parameters']['t_step']
+            del sdict['parameters']['t_step']
+        return sdict
+
+    def file_writer_helper_func(self, dict, filename):
```

Comment:
I'm a bit confused, can you not just use the f90ml writing methods instead of doing all of this?

Reply:
I tried using the f90ml writing methods but it returned a format of namelists that ShengBTE couldn't read